check-log/lib/check-log.go

```diff
@@ -316,9 +332,18 @@ func (opts *logOpts) searchLog(logFile string) (int64, int64, string, error) {
 	return warnNum, critNum, errLines, nil
 }
 
-func (opts *logOpts) searchReader(rdr io.Reader) (warnNum, critNum, readBytes int64, errLines string, err error) {
-	r := bufio.NewReader(rdr)
-	for {
+func newBufferedReader(r io.Reader) *bufio.Reader {
+	return bufio.NewReader(r)
+}
+
+func (opts *logOpts) searchReader(ctx context.Context, rdr io.Reader) (warnNum, critNum, readBytes int64, errLines string, err error) {
+	newReader := opts.testHookNewBufferedReader
+	if newReader == nil {
+		newReader = newBufferedReader
+	}
+
+	r := newReader(rdr)
+	for ctx.Err() == nil {
```

Comment:
With this implementation, searching a file is not interrupted after this `ctx.Err()` check.
It's better to stop scanning on context cancel (= signal), even after this check.

Reply:
Oops, I misread `for` as `if`...